saleor/checkout/fetch.py

```diff
@@ -250,6 +252,17 @@ def fetch_checkout_lines(
             checkout, product, variant_channel_listing, product_channel_listing_mapping
         ):
             unavailable_variant_pks.append(variant.pk)
+            if not skip_lines_with_unavailable_variants:
```

Comment:
What is the reason for having this?
Question: Did you check your logic with the scenario when checkout has some lines that are unavailable?

Reply:
This flag was added because in some places where `fetch_checkout_lines` is used returning `CheckoutLineInfo` for unavailable variant breaks things for example in stripe/aydien plugins